src/BankAccount.java

```diff
@@ -0,0 +1,166 @@
+import java.util.HashMap;
+
+class Account {
+    int amount;
+    double interest;
+    String type;
+    int overdraft;
+}
+
+public class BankAccount {
+
+    // Private for Map of Stocks
+    private HashMap<String, Account> accounts;
+    private int size;
+
+    private void createAccount() {
+        this.accounts = new HashMap<String, Account>();
+        this.size = 0;
+    }
+
+    public BankAccount() {
+        this.createAccount();
+    }
+
+    public final void addAccount(String account, int deposit, double interest,
+            String type, int overdraft) {
```

Comment:
In the future, you might set type to a enum. I imagine there are only so many types of accounts you would be willing to support. Enums avoid the issue of typos.